### PR TITLE
WIP: Support dark mode on the site

### PR DESCRIPTION
**Pull request by redonkulus**
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

---

Working on updating the site to support dark mode. 

- Moved most of the hardcoded colors to a new colors.css sheet.
- A toggle to flip between scheme's will be added in a follow up PR.

### CODE REVIEW

File: app/assets/css/custom.css [hunk 1]
```diff
@@ -71,6 +71,11 @@ li.selected a {
     background: rgba(2, 128, 174, 0.3);
 }
 
+#nav li:hover a,
+#nav li.selected a {
+    color: #0280ae99;
```
Comment (carriemorrison) [inline app/assets/css/custom.css:76]: color var?

File: app/assets/css/custom.css [hunk 2]
```diff
@@ -217,3 +195,13 @@ pre code {
     height: 16px;
     width: 16px;
 }
+
+/* next.js logo needs to invert color based on color scheme  */
```
Comment (carriemorrison) [inline app/assets/css/custom.css:199]: 👍

File: app/assets/css/mq.css [hunk 3]
```diff
@@ -7,16 +7,16 @@
      */
     .menu-on #nav {
         animation: slide 0.3s ease-out;
+        background: var(--color-bg);
         box-shadow: 1px 0 9px 1px #555;
```
Comment (carriemorrison) [inline app/assets/css/mq.css:11]: does this need a var?

File: app/assets/css/repl.css [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
+.repl pre,
+.repl textarea {
     box-shadow: none;
 }
 
-textarea {
+.repl textarea {
     background: none;
     border: 0;
     color: #2a2734;
```
Comment (carriemorrison) [inline app/assets/css/repl.css:11]: var needed?

File: app/assets/css/syntax.css [hunk 5]
```diff
@@ -1,227 +1,357 @@
...
-    background: url(https://s.yimg.com/os/acss/images/pre-background.177ebfe5.gif);
-}
-
 /* https://github.com/jwarby/jekyll-pygments-themes/blob/master/github.css */
 .highlight {
     margin: 0 0 1em;
 }
+
 .highlight .hll {
     background-color: #ffffcc;
```
Comment (carriemorrison) [inline app/assets/css/syntax.css:7]: var?

File: app/assets/css/syntax.css [hunk 6]
```diff
@@ -1,227 +1,357 @@
...
+
+/* Generic.Output */
 .highlight .go {
     color: #888888;
-} /* Generic.Output */
+}
+
+/* Generic.Prompt */
 .highlight .gp {
     color: #555555;
```
Comment (carriemorrison) [inline app/assets/css/syntax.css:95]: var?

File: app/assets/css/syntax.css [hunk 7]
```diff
@@ -1,227 +1,357 @@
...
+
+/* Generic.Strong */
 .highlight .gs {
     font-weight: bold;
-} /* Generic.Strong */
+}
+
+/* Generic.Subheading */
 .highlight .gu {
     color: #aaaaaa;
```
Comment (carriemorrison) [inline app/assets/css/syntax.css:105]: var?

File: atomizer.config.js [hunk 8]
```diff
@@ -7,11 +7,19 @@ module.exports = {
     },
     content: ['app/components/', 'app/assets/js', 'docs/'],
     custom: {
-        borderTopLight: '1px solid rgba(2, 128, 174, 0.3)',
+        borderTopLight: '1px solid var(--color-border-light)',
         boxColorLight: 'rgba(2, 128, 174, 0.05)',
         boxColorDarker: 'rgba(2, 128, 174, 0.2)',
```
Comment (carriemorrison) [inline atomizer.config.js:12]: these two needed?

File: atomizer.config.js [hunk 9]
```diff
@@ -7,11 +7,19 @@ module.exports = {
...
-        boxShadowLight: '0 2px 3px rgba(0, 0, 0, 0.2)',
-        brandColor: '#0280ae',
+        hamburgerImage: 'url(https://s.yimg.com/os/acss/images/menu.d6610eea.png)',
```
Comment (carriemorrison) [inline atomizer.config.js:13]: do we need darkmode burger icon? maybe we should move to SVG then can just use the color var on fill...

File: docs/content.md [hunk 10]
```diff
@@ -91,10 +91,10 @@ Atomizer parses any text document you define in your `atomizer.config.js` config
 
 ```html
 <div class="D(f) Jc(sb)">
-    <div class="W(100px) H(100px) Lh(100px) Fz(30px) Ta(c) Bgc(#0280ae.5)">Box 1</div>
```
Comment (carriemorrison) [inline docs/content.md:94]: we could look at adding var opacity support for  atomizer.

File: docs/index.md [hunk 11]
```diff
@@ -137,26 +137,26 @@ custom: {
     </div>
     <div class="Fl(start) W(60%) Cl(b) Fl(n)--xs W(a)--xs">
 {% highlight html %}
-<div class="Bd Bgc(#0280ae):h C(#0280ae) C(#fff):h P(20px)">
+<div class="Bgc(#0280ae):h C(#0280ae) C(#fff):h">
```
Comment (carriemorrison) [inline docs/index.md:140]: assume these are ok...

File: docs/thinking-in-atomic.md [hunk 12]
```diff
@@ -14,7 +14,7 @@ title: Thinking in Atomic
 Adopting an Atomic CSS methodology addresses common CSS challenges:
 
 <dl class="Mb(30px)">
-    <dt class="Fs(i) C(#000)">Changes are predictable</dt>
+    <dt class="Fs(i) C(--color-emphasis)">Changes are predictable</dt>
     <dd class="Mstart(20px) Mt(5px)">Because of the single responsibility principle (one class == one style) it is easy to predict what removing or adding a class will do.</dd>
     <dt class="Fs(i) C(#000)">Scope is limited</dt>
```
Comment (carriemorrison) [inline docs/thinking-in-atomic.md:19]: `--color-emphasis` ?

**Review by carriemorrison — DISMISSED** [review]
Minor comments, as discussed we could also add `color-scheme: dark;` for scrollbars in darkmode..

File: app/assets/css/repl.css (hunk 4 above)
Comment (redonkulus) [reply app/assets/css/repl.css:11]: For colors that are only used once, I just left them hardcoded

File: app/assets/css/custom.css (hunk 1 above)
Comment (redonkulus) [reply app/assets/css/custom.css]: updated in latest commit

File: app/assets/css/syntax.css (hunk 5 above)
Comment (redonkulus) [reply app/assets/css/syntax.css:7]: Only used once, so left hardcoded. Did that for many of the syntax colors.

File: app/assets/css/syntax.css (hunk 7 above)
Comment (redonkulus) [reply app/assets/css/syntax.css:105]: nope, only used once here.

File: atomizer.config.js (hunk 8 above)
Comment (redonkulus) [reply atomizer.config.js]: they have been removed in a recent commit

File: atomizer.config.js (hunk 9 above)
Comment (redonkulus) [reply atomizer.config.js:10]: 👍 yup, I want to move to svg and animate to close icon

File: docs/content.md (hunk 10 above)
Comment (redonkulus) [reply docs/content.md:94]: what would the syntax look like for that?

File: docs/index.md (hunk 11 above)
Comment (redonkulus) [reply docs/index.md:167]: I left the examples with hardcoded hex to make it clearer to users. I didn't want to complicate it with CSS vars.

File: docs/thinking-in-atomic.md (hunk 12 above)
Comment (redonkulus) [reply docs/thinking-in-atomic.md]: Good find! We have custom css to style the `<dt>` tag so the `C()` class isn't actually needed. Removed in https://github.com/acss-io/atomizer/pull/505/commits/68ae81e63bbde3fcc9dc07626fc153884b6c7a97

File: app/assets/css/mq.css (hunk 3 above)
Comment (redonkulus) [reply app/assets/css/mq.css]: Removed here https://github.com/acss-io/atomizer/pull/505/commits/68ae81e63bbde3fcc9dc07626fc153884b6c7a97

**Comment by redonkulus** [comment]
> Minor comments, as discussed we could also add color-scheme: dark; for scrollbars in darkmode..

Added a fix for this too

File: docs/content.md (hunk 10 above)
Comment (carriemorrison) [reply docs/content.md:94]: `Bgc(--text-foo.5)`

Comment (redonkulus) [reply docs/content.md:94]: Seems kind of odd to add the value onto the variable, versus just making it a variable.

**Pull request merged**